src/test/java/teammates/test/cases/search/InstructorSearchTest.java

```diff
@@ -0,0 +1,90 @@
+package teammates.test.cases.search;
+
+import java.util.Arrays;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.InstructorSearchResultBundle;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.storage.api.InstructorsDb;
+import teammates.storage.search.InstructorSearchDocument;
+import teammates.storage.search.InstructorSearchQuery;
+import teammates.test.driver.AssertHelper;
+
+/**
+ * SUT: {@link InstructorsDb}, {@link InstructorSearchDocument},
+ * {@link InstructorSearchQuery}.
+ */
+public class InstructorSearchTest extends BaseSearchTest {
+    @Test
+    public void allTests() {
+        InstructorsDb instructorsDb = new InstructorsDb();
+
+        InstructorAttributes ins1InCourse1 = dataBundle.instructors.get("instructor1OfCourse1");
+        InstructorAttributes ins2InCourse1 = dataBundle.instructors.get("instructor2OfCourse1");
+        InstructorAttributes ins1InCourse2 = dataBundle.instructors.get("instructor1OfCourse2");
+        InstructorAttributes ins2InCourse2 = dataBundle.instructors.get("instructor2OfCourse2");
+        InstructorAttributes insInArchivedCourse = dataBundle.instructors.get("instructorOfArchivedCourse");
+        InstructorAttributes insInUnregCourse = dataBundle.instructors.get("instructor5");
+
+        ______TS("success: search for instructors in whole system; query string does not match anyone");
+
+        InstructorSearchResultBundle results =
+                instructorsDb.searchInstructorsInWholeSystem("non-existent");
+        verifySearchResults(results);
+
+        ______TS("success: search for instructors in whole system; query string matches some instructors");
```

Comment:
There should be more test cases here. As indicated in `InstructorSearchDocument.java`
``` java
    @Override
    protected Document toDocument() {
        
        String delim = ",";
        
        // produce searchableText for this instructor document:
        // it contains courseId, courseName, instructorName, instructorEmail, instructorGoogleId, instructorRole
        String searchableText = instructor.courseId + delim
                                + (course == null ? "" : course.getName()) + delim
                                + instructor.name + delim
                                + instructor.email + delim
                                + (instructor.googleId == null ? "" : instructor.googleId) + delim
                                + instructor.role + delim
                                + instructor.displayedName;
        
// ...
    }
```
Instructor's `courseId` `courseName` `name` `email` `googleId` `role` `displayedName` will be searchable texts. Suggest to have test cases on this.